cpp/src/arrow/filesystem/azurefs.h

```diff
@@ -154,6 +154,9 @@ class ARROW_EXPORT AzureFileSystem : public FileSystem {
 
   explicit AzureFileSystem(std::unique_ptr<Impl>&& impl);
 
+ protected:
+  AzureFileSystem* ForceCachedHierarchicalNamespaceSupport(int hns_support);
```

Comment:
I'm not sure why this is returning a `AzureFileSystem*`. This is not an idiom we use in general.

Reply:
I can take it out and restrict it to the code in tests.